src/util/util.h

```diff
@@ -250,11 +250,11 @@ struct sss_names_ctx {
     sss_regexp_t *re;
 };
 
-#define SSS_DEFAULT_RE "(?P<name>[^@]+)@?(?P<domain>[^@]*$)"
+#define SSS_DEFAULT_RE "^((?P<name>.+)@(?P<domain>[^@]*)|(?P<name>[^@]+))$"
```

Comment:
IIUC, this regex accepts the expression `name@` as valid and considers the domain is an empty string. Shouldn't we have a `+` instead of the `*` in the domain part (`?P<domain>[^@]*`) ?

Reply:
@aplopez - yes you are right but the old expression allowed the same thing.

It looks like we also have tests for that trailing/leading `@` and they should fail (but they do not)

https://github.com/SSSD/sssd/blob/9e9d5825d1c331bc8bfb28a1d2ac456a085ba380/src/tests/cmocka/test_fqnames.c#L507 ).

Thanks for pointing that out.

Reply:
@aplopez - the tests were using only IPA/AD regexp in this negative test. I extended them to have this tested also for default regular expression.

The `*` is now replaced by `+` as you suggested. Please take a look.